### ISSUE TITLE
Specifying "extras" in pip / requirements.txt results in false negative

### ISSUE THREAD

**Issue by tlbjr** [aquasecurity/trivy#4101]
## Description

Python's (pip) celery 4.4.7 has a high vuln. This was identified with MS' Defender for Cloud.
We have trivy vulns scanning in our CI/CD, but it wasn't found because we have "extras" specified as acceptable to pip / requirements.txt.
```
celery[redis, pytest]==4.4.7  # Will not find a high vuln
#celery==4.4.7  # Will find a high vuln as expected
starlette==0.17.1
```

## What did you expect to happen?
At the very least, ignore all extras (consistent with no transitive dependency checking in trivy's pip/requirements.txt support) and compare the base package and version to the list of known vulns.

## What happened instead?
The library and version was not matched against any known vulns. I have an example medium vuln included to show a scan is at least taking place.

## Output of run with `--debug`:

```
$ cat requirements.txt
celery[redis, pytest]==4.4.7  # Will not find a high vuln
#celery==4.4.7  # Will find a high vuln as expected
starlette==0.17.1
$ trivy fs requirements.txt --debug
2023-04-20T08:37:42.670-0700    DEBUG   Severities: ["UNKNOWN" "LOW" "MEDIUM" "HIGH" "CRITICAL"]
2023-04-20T08:37:42.672-0700    DEBUG   cache dir:  /home/user/.cache/trivy
2023-04-20T08:37:42.673-0700    DEBUG   DB update was skipped because the local DB is the latest
2023-04-20T08:37:42.673-0700    DEBUG   DB Schema: 2, UpdatedAt: 2023-04-20 12:07:40.789768334 +0000 UTC, NextUpdate: 2023-04-20 18:07:40.789767934 +0000 UTC, DownloadedAt: 2023-04-20 14:08:45.316528344 +0000 UTC
2023-04-20T08:37:42.673-0700    INFO    Vulnerability scanning is enabled
2023-04-20T08:37:42.673-0700    DEBUG   Vulnerability type:  [os library]
2023-04-20T08:37:42.673-0700    INFO    Secret scanning is enabled
2023-04-20T08:37:42.673-0700    INFO    If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2023-04-20T08:37:42.673-0700    INFO    Please see also https://aquasecurity.github.io/trivy/v0.40/docs/secret/scanning/#recommendation for faster secret detection
2023-04-20T08:37:42.673-0700    DEBUG   No secret config detected: trivy-secret.yaml
2023-04-20T08:37:42.673-0700    DEBUG   Walk the file tree rooted at 'requirements.txt' in parallel
2023-04-20T08:37:42.676-0700    DEBUG   OS is not detected.
2023-04-20T08:37:42.676-0700    DEBUG   Detected OS: unknown
2023-04-20T08:37:42.676-0700    INFO    Number of language-specific files: 1
2023-04-20T08:37:42.676-0700    INFO    Detecting pip vulnerabilities...
2023-04-20T08:37:42.676-0700    DEBUG   Detecting library vulnerabilities, type: pip, path: requirements.txt

requirements.txt (pip)

Total: 1 (UNKNOWN: 0, LOW: 0, MEDIUM: 1, HIGH: 0, CRITICAL: 0)

┌───────────┬─────────────────────┬──────────┬───────────────────┬───────────────┬───────────────────────────────────────────────────────────┐
│  Library  │    Vulnerability    │ Severity │ Installed Version │ Fixed Version │                           Title                           │
├───────────┼─────────────────────┼──────────┼───────────────────┼───────────────┼───────────────────────────────────────────────────────────┤
│ starlette │ GHSA-74m5-2c7w-9w3x │ MEDIUM   │ 0.17.1            │ 0.25.0        │ MultipartParser denial of service with too many fields or │
│           │                     │          │                   │               │ files                                                     │
│           │                     │          │                   │               │ https://github.com/advisories/GHSA-74m5-2c7w-9w3x         │
└───────────┴─────────────────────┴──────────┴───────────────────┴───────────────┴───────────────────────────────────────────────────────────┘
```

Run without extras:
```
$ cat requirements.txt
#celery[redis, pytest]==4.4.7  # Will not find a high vuln
celery==4.4.7  # Will find a high vuln as expected
starlette==0.17.1
$ trivy fs requirements.txt --debug
2023-04-20T08:42:24.972-0700    DEBUG   Severities: ["UNKNOWN" "LOW" "MEDIUM" "HIGH" "CRITICAL"]
2023-04-20T08:42:24.975-0700    DEBUG   cache dir:  /home/user/.cache/trivy
2023-04-20T08:42:24.976-0700    DEBUG   DB update was skipped because the local DB is the latest
2023-04-20T08:42:24.976-0700    DEBUG   DB Schema: 2, UpdatedAt: 2023-04-20 12:07:40.789768334 +0000 UTC, NextUpdate: 2023-04-20 18:07:40.789767934 +0000 UTC, DownloadedAt: 2023-04-20 14:08:45.316528344 +0000 UTC
2023-04-20T08:42:24.976-0700    INFO    Vulnerability scanning is enabled
2023-04-20T08:42:24.976-0700    DEBUG   Vulnerability type:  [os library]
2023-04-20T08:42:24.976-0700    INFO    Secret scanning is enabled
2023-04-20T08:42:24.976-0700    INFO    If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2023-04-20T08:42:24.976-0700    INFO    Please see also https://aquasecurity.github.io/trivy/v0.40/docs/secret/scanning/#recommendation for faster secret detection
2023-04-20T08:42:24.976-0700    DEBUG   No secret config detected: trivy-secret.yaml
2023-04-20T08:42:24.976-0700    DEBUG   Walk the file tree rooted at 'requirements.txt' in parallel
2023-04-20T08:42:24.983-0700    DEBUG   OS is not detected.
2023-04-20T08:42:24.983-0700    DEBUG   Detected OS: unknown
2023-04-20T08:42:24.983-0700    INFO    Number of language-specific files: 1
2023-04-20T08:42:24.983-0700    INFO    Detecting pip vulnerabilities...
2023-04-20T08:42:24.983-0700    DEBUG   Detecting library vulnerabilities, type: pip, path: requirements.txt

requirements.txt (pip)

Total: 2 (UNKNOWN: 0, LOW: 0, MEDIUM: 1, HIGH: 1, CRITICAL: 0)

┌───────────┬─────────────────────┬──────────┬───────────────────┬───────────────┬───────────────────────────────────────────────────────────┐
│  Library  │    Vulnerability    │ Severity │ Installed Version │ Fixed Version │                           Title                           │
├───────────┼─────────────────────┼──────────┼───────────────────┼───────────────┼───────────────────────────────────────────────────────────┤
│ celery    │ CVE-2021-23727      │ HIGH     │ 4.4.7             │ 5.2.2         │ celery: stored command injection vulnerability may allow  │
│           │                     │          │                   │               │ privileges escalation                                     │
│           │                     │          │                   │               │ https://avd.aquasec.com/nvd/cve-2021-23727                │
├───────────┼─────────────────────┼──────────┼───────────────────┼───────────────┼───────────────────────────────────────────────────────────┤
│ starlette │ GHSA-74m5-2c7w-9w3x │ MEDIUM   │ 0.17.1            │ 0.25.0        │ MultipartParser denial of service with too many fields or │
│           │                     │          │                   │               │ files                                                     │
│           │                     │          │                   │               │ https://github.com/advisories/GHSA-74m5-2c7w-9w3x         │
└───────────┴─────────────────────┴──────────┴───────────────────┴───────────────┴───────────────────────────────────────────────────────────┘
```

## Output of `trivy -v`:

```
Version: 0.40.0
Vulnerability DB:
  Version: 2
  UpdatedAt: 2023-04-20 12:07:40.789768334 +0000 UTC
  NextUpdate: 2023-04-20 18:07:40.789767934 +0000 UTC
  DownloadedAt: 2023-04-20 14:08:45.316528344 +0000 UTC
```

**Comment by DmitriyLewen** [comment]
Hello @tlbjr 
Thanks for your report!

It is duplicate of #4095. Lets talk about this problem there.
